modin/pandas/test/test_groupby.py

```diff
@@ -2099,3 +2100,107 @@ def test_mean_with_datetime(by_func):
 
     modin_df, pandas_df = create_test_dfs(data)
     eval_general(modin_df, pandas_df, lambda df: df.groupby(by=by_func(df)).mean())
+
+
+@pytest.mark.parametrize(
+    "modin_df_recipe",
+    ["non_lazy_frame", "frame_with_deferred_index", "lazy_frame"],
+)
+def test_groupby_on_empty_data(modin_df_recipe):
+    class ModinDfConstructor:
+        def __init__(self, recipie, df_kwargs):
+            self._recipie = recipie
+            self._mock_obj = None
+            self._df_kwargs = df_kwargs
+
+        def non_lazy_frame(self):
+            return pd.DataFrame(**self._df_kwargs)
+
+        def frame_with_deferred_index(self):
+            df = pd.DataFrame(**self._df_kwargs)
+            try:
+                # The frame would stop being lazy once index computation is triggered
+                df._query_compiler._modin_frame._index_cache = None
```

Comment:
Does this trigger index computation for HDK even if we use _index_cache rather than index property?

Reply:
No. HDK triggers execution when the [`._get_index()`](https://github.com/dchigarev/modin/blob/f6f2f36dbcfcde0cdf271dc4eccb7653d0e32f99/modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/dataframe.py#L1889) is called, which happens only when we access the `.index` property.